test/unit/kubernetes-deploy/options_helper_test.rb

```diff
@@ -0,0 +1,114 @@
+# frozen_string_literal: true
+require 'test_helper'
+require 'tempfile'
+
+class OptionsHelperTest < KubernetesDeploy::TestCase
+  def test_single_template_dir_only
+    KubernetesDeploy::OptionsHelper.with_consolidated_template_dir([fixture_path('hello-cloud')]) do |template_dir|
+      assert_equal(fixture_path('hello-cloud'), template_dir)
+    end
+  end
+
+  def test_with_default_env_var
+    old_env = ENV["ENVIRONMENT"]
+    ENV["ENVIRONMENT"] = "test"
+    KubernetesDeploy::OptionsHelper.with_consolidated_template_dir([]) do |template_dir|
+      assert_equal(template_dir, File.join("config", "deploy", "test"))
+    end
+  ensure
+    ENV["ENVIRONMENT"] = old_env
+  end
+
+  def test_multiple_template_dirs
+    template_dirs = [fixture_path('hello-cloud'), fixture_path('partials')]
+
+    KubernetesDeploy::OptionsHelper.with_consolidated_template_dir(template_dirs) do |template_dir|
+      fixture_path_entries = template_dirs.collect { |dir| Dir.entries(dir) }.flatten.uniq
+      template_dir_entries = Dir.entries(template_dir)
+      assert_equal(fixture_path_entries.length, template_dir_entries.length)
+      fixture_path_entries.each do |fixture|
+        refute(template_dir_entries.select { |s| s.include?(fixture) }.empty?)
+      end
+    end
+  end
+
+  def test_missing_template_dir_raises
+    assert_raises(KubernetesDeploy::OptionsHelper::OptionsError) do
+      KubernetesDeploy::OptionsHelper.with_consolidated_template_dir([]) do
+      end
+    end
+  end
+
+  def test_template_dir_with_stdin
+    old_stdin = $stdin
+
+    input = Tempfile.open("kubernetes_deploy_test")
+    File.open(File.join(fixture_path('for_unit_tests'), "service_test.yml"), 'r') do |f|
+      input.print(f.read)
+    end
+    input.rewind
+    $stdin = input
```

Comment:
Can you write to `$stdin` instead of abusing a global?

Reply:
From what I gather, no. The best I can do is ensure that `old_stdin` is reset at the end of the test. I'm not sure how I could even write to `$stdin` without changing the descriptor itself, regardless.